kernel_module/uio/uio_ivshmem.c

```diff
@@ -115,14 +140,26 @@ static int request_msix_vectors(struct ivshmem_info *ivs_info, int nvectors)
 			free_msix_vectors(ivs_info, i - 1);
 			goto error;
 		}
+#else
+		err = request_irq(pci_irq_vector(ivs_info->dev,
+			ivs_info->msix_entries[i].entry), ivshmem_msix_handler, 0,
+			ivs_info->msix_names[i], ivs_info->uio);
 
+		if (err < 0) {
+			free_msix_vectors(ivs_info, i - 1);
+			goto error;
+		}
+#endif
 	}
 
 	return 0;
 error:
 	kfree(ivs_info->msix_entries);
 	kfree(ivs_info->msix_names);
 	ivs_info->nvectors = 0;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,8,0)
+	pci_free_irq_vectors(ivs_info->dev);
+#endif
 	return err;
```

Comment:
Something similar will also have to go into ivshmem_pci_remove. I think that should be called when you unbind the device from the driver or when you rmmod the module.